src/NLog/Internal/AssemblyHelpers.cs

```diff
@@ -102,9 +103,36 @@ public static Assembly LoadFromName(string assemblyName)
             Assembly assembly = assemblyPart.Load(stream.Stream);
             return assembly;
 #else
-            Assembly assembly = Assembly.Load(assemblyName);
-            return assembly;
+            try
+            {
+                Assembly assembly = Assembly.Load(AssemblyName.GetAssemblyName(assemblyName));
+                return assembly;
+            }
+            catch (FileNotFoundException)
+            {
+                var name = new AssemblyName(assemblyName);
+                var loadedAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(domainAssembly => CheckAssembly(name, domainAssembly.GetName()));
+                if (loadedAssembly != null)
+                    return loadedAssembly;
+                else throw;
+            }
+#endif
+        }
+
+        static bool CheckAssembly(AssemblyName expected, AssemblyName actual)
```

Comment:
Proposal name: IsSameAssembly

Reply:
I admit, my name is not good. But in my opinion 'IsSameAssembly' is also misleading, because it will also return true, if `expected` does not have a Version or Culture specified, but `actual` has. I was thinking of 'MeetsCriteria' or simply 'IsRequested'? During this I would also rename 'expected'-parameter to 'requested'.
Because we are in a static class, I could also implement it as an Extension Method, so we could call it like `domainAssembly.MeetsCriteria(name)`

Reply:
IsAssemblyMatch?

I use match for fuzzy matches. :) (Otherwise "equals")

Reply:
renamed